service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -243,6 +252,15 @@ private WdsRecordResponseDetails fetchWdsRecords(
         recordIdsWithError.put(recordId, e.getMessage());
       }
     }
+    micrometerMetrics.stopTimer(
+        wdsFetchRecordsSample,
+        "wds_fetch_records_timer",
+        "run_set_id",
+        runSet.runSetId().toString(),
+        "recordResponses",
+        recordResponses.toString(),
+        "recordIdsWithError",
+        recordIdsWithError.toString());
```

Comment:
TOL: Maybe this particular timer metric can be split into 2?
- time to fetch all records data from WDS successfully
- timing for unsuccessful fetch

Reply:
yeah that's a good thought - in another metric I'm adding here, there is a field indicating success/failure, but we could instead use a different metric name (for that metric and this one)

Reply:
Actually, as I look at this again: I think we should talk about how we want to do this in more detail. Timing unsuccessful fetches will change the nature of this metric from "time to request records in a given submission" to "time to request any given record, split into separate success/failure metrics"

That's a fine change, but it's different enough that I think we should discuss -- I'll set up a time later today

Reply:
After discussing with Saloni: add "percentage of records successfully retrieved", instead of binary success/failure